pxtblocks/blocklyloader.ts

```diff
@@ -913,21 +1058,168 @@ namespace pxt.blocks {
     }
+
+    function initLogic() {
+        let msg:any = Blockly.Msg;
+
+        // builtin controls_if
+        msg.CONTROLS_IF_MSG_IF = lf("_logic_if");
+        msg.CONTROLS_IF_MSG_THEN = lf("_logic_then");
+        msg.CONTROLS_IF_MSG_ELSE = lf("_logic_else");
+        msg.CONTROLS_IF_MSG_ELSEIF = lf("_logic_elseif");
+        msg.CONTROLS_IF_TOOLTIP_1 = lf("If a value is true, then do some statements.");
+        msg.CONTROLS_IF_TOOLTIP_2 = lf("If a value is true, then do the first block of statements. Otherwise, do the second block of statements.");
+        msg.CONTROLS_IF_TOOLTIP_3 = lf("If the first value is true, then do the first block of statements. Otherwise, if the second value is true, do the second block of statements.");
+        msg.CONTROLS_IF_TOOLTIP_4 = lf("If the first value is true, then do the first block of statements. Otherwise, if the second value is true, do the second block of statements. If none of the values are true, do the last block of statements.");
+        installHelpResources(
+            'controls_if',
+            lf("a conditional statement"),
+            undefined,
+            "blocks/logic/if"
+        );
+
+        // builtin logic_compare
+        Blockly.Blocks['logic_compare'].init = function() {
+            let OPERATORS = [
+                [lf("=="),  'EQ'],  // =
+                [lf("_op_neq"), 'NEQ'], // ≠
+                [lf("_op_lt"),  'LT'],  // <  (ltr)
+                [lf("_op_lte"), 'LTE'], // ≦ (ltr)
+                [lf("_op_gt"),  'GT'],  // >  (ltr)
+                [lf("_op_gte"), 'GTE']  // ≧ (ltr)
+            ];
+            this.setColour(blockColors['logic']);
+            this.setOutput(true, 'Boolean');
+            this.appendValueInput('A');
+            this.appendValueInput('B')
+                .appendField(new Blockly.FieldDropdown(OPERATORS), 'OP');
+            this.setInputsInline(true);
+            this.prevBlocks_ = [null, null];
+
+            let TOOLTIPS:any = {
+                'EQ': lf("Return true if both inputs equal each other."),
+                'NEQ': lf("Return true if both inputs are not equal to each other."),
+                'LT': lf("Return true if the first input is smaller than the second input."),
+                'LTE': lf("Return true if the first input is smaller than or equal to the second input."),
+                'GT': lf("Return true if the first input is greater than the second input."),
+                'GTE': lf("Return true if the first input is greater than or equal to the second input.")
+            };
+
+            var thisBlock = this;
+            setHelpResources(this,
+                'logic_compare',
+                lf("comparing two numbers"),
+                function() {
+                    return TOOLTIPS[thisBlock.getFieldValue('OP')];
+                },
+                '/blocks/logic/boolean'
+            );
+        };
+
+        // builtin logic_operation
+        msg.LOGIC_OPERATION_AND = lf("&&");
```

Comment:
On a second thought, ``&&`` and ``||`` are cryptic for kids. We should stick to the natural language description: ``lf("and")`` and ``lf("or")``. Sorry for the confusion.